packages/itwinui-react/src/core/ComboBox/ComboBox.tsx

```diff
@@ -481,22 +494,26 @@ export const ComboBox = React.forwardRef(
               .join(', '),
           );
 
-          // Clear the filter whenever an item is selected
-          setInputValue('');
+          // If clearFilterOnOptionToggle, clear the filter
+          setIsInputDirty(!clearFilterOnOptionToggle);
```

Comment:
Should this be written as an `if` condition? Because otherwise, the comment is a lie by omission (it's also calling `setIsInputDirty(true)` when `clearFilterOnOptionToggle` is false).

```suggestion
          if (clearFilterOnOptionToggle) {
            setIsInputDirty(false);
          }
```

This can probably be combined with the if-condition below.